tests/test_tools.py

```diff
@@ -55,6 +55,19 @@ async def invalid_tool(ctx: RunContext[None]) -> str:  # pragma: no cover
     )
 
 
+def test_builtin_tool_registration():
+    """
+    Test that built-in functions can be registered as tools.
+    """
+
+    with pytest.raises(UserError, match='Error generating schema for max:'):
```

Comment:
Can we include the full error message here? I want to see the `{e}` in `Error getting signature for {function.__qualname__}: {e}`, to make sure it's clear that the problem is that it's a built-in. 

Also, we don't need to repeat the function name there if it's already in the `Error generating schema for max:` line